controllers/servingruntime_controller.go

```diff
@@ -509,25 +509,55 @@ func (r *ServingRuntimeReconciler) SetupWithManager(mgr ctrl.Manager,
 }
 
 func (r *ServingRuntimeReconciler) requestsForRuntimes(namespace string,
-	filter func(*kserveapi.ServingRuntime) bool) []reconcile.Request {
+	filter func(string) bool) []reconcile.Request {
 	var opts []client.ListOption
 	if namespace != "" {
 		opts = []client.ListOption{client.InNamespace(namespace)}
 	}
-	list := &kserveapi.ServingRuntimeList{}
-	if err := r.Client.List(context.TODO(), list, opts...); err != nil {
+	runtimes := &kserveapi.ServingRuntimeList{}
+	if err := r.Client.List(context.TODO(), runtimes, opts...); err != nil {
 		r.Log.Error(err, "Error listing ServingRuntimes to reconcile", "namespace", namespace)
 		return []reconcile.Request{}
 	}
-	requests := make([]reconcile.Request, 0, len(list.Items))
-	for i := range list.Items {
-		rt := &list.Items[i]
-		if filter == nil || filter(rt) {
-			requests = append(requests, reconcile.Request{
-				NamespacedName: types.NamespacedName{Name: rt.Name, Namespace: rt.Namespace},
-			})
+
+	srnns := make(map[types.NamespacedName]struct{})
```

Comment:
There's no need for this in the `!r.EnableCSRWatch` case so maybe move it within the `if` block and just handle the SRs like they are now if CSRs are disabled? i.e. different loop over SR list in both if and else cases.